src/main/java/ch/njol/skript/expressions/ExprBookAuthor.java

```diff
@@ -56,7 +56,9 @@ protected String getPropertyName() {
 	
 	@Nullable
 	@Override
-	public String convert(ItemStack itemStack) {
+	public String convert(ItemType item) {
+		ItemStack itemStack = item.getRandom();
```

Comment:
You can do this check and get the item meta with the ItemType so there's no need to call getRandom.